### ISSUE TITLE
loss计算

### ISSUE THREAD

**Issue by CamillerFerros** [DeepMed-Lab-ECNU/BCP#25]
def _one_hot_mask_encoder(self, input_tensor):
        tensor_list = []
        for i in range(self.n_classes):
            temp_prob = input_tensor * i == i * torch.ones_like(input_tensor)
            tensor_list.append(temp_prob)
        output_tensor = torch.cat(tensor_list, dim=1)
        return output_tensor.float()
您好，请问这段代码进行计算loss的时候是否存在问题，当i=0即计算背景的时候非裁剪的部分也被算了进来，i=1,2,3时计算的都是裁剪后的loss，但是i=0的时候背景算的是整张图的loss而非裁剪部分。是否计算是有问题的呢。

**Comment by byhwhite** [comment]
您好！非常感谢您的关注和这个问题！这个loss确实是有问题的！我们连夜加急更改了loss并跑出了结果，更改loss后的结果确实比原本的结果更好一些！再次感谢您的反馈！
<img width="1365" alt="image" src="https://github.com/DeepMed-Lab-ECNU/BCP/assets/51234133/1122a2ad-b384-42f2-a03d-6a39318cd433">

**Comment by byhwhite** [comment]
该loss已在github中更新，原代码实现有些ugly😑😑。若代码仍有问题期待您的反馈！

**Comment by CamillerFerros** [comment]
你好，请问你们的工作中，网络模型除了使用unet还尝试过其他的网络框架吗，结果与unet相比怎么样，期待您的回复。

**Comment by byhwhite** [comment]
> 你好，请问你们的工作中，网络模型除了使用unet还尝试过其他的网络框架吗，结果与unet相比怎么样，期待您的回复。

你好，为了与其他方法公平比较，我们在主要实验中backbone皆与其他方法保持一致。但是在rebuttal中，审稿人好奇我们的方法在较大体量的网络中性能如何，所以我们将BCP与其他对比方法在较大的网络中（更深的Vnet）的性能进行了对比。结果是，在半监督setting中，较大体量的网络更容易过拟合到labeled data上，所有方法训练好的网络都展现出了对labeled set更高的和在unlabeled set和test set中更差的测试性能，但是BCP仍然保持所有方法中性能差距最小的表现（即使用更大模型后，BCP与其他方法的定性对比仍如论文中图2所示）。希望我们的结论对你有所帮助。

**Comment by CamillerFerros** [comment]
您好，请问您修改了loss之后，在ACDC数据集上的7个标签的最终结果是多少，有修改loss之后的其他结果吗，期待您的回复

**Comment by byhwhite** [comment]
> 您好，请问您修改了loss之后，在ACDC数据集上的7个标签的最终结果是多少，有修改loss之后的其他结果吗，期待您的回复

您好，我们暂时没有跑完整个实验，您若想与我们的方法进行比对可以按照论文中的结果。
